BUILD.ghc

```diff
@@ -2,6 +2,7 @@
 package(default_visibility = [ "//visibility:public" ])
 
 load("@io_tweag_rules_haskell//haskell:haskell.bzl", "haskell_toolchain")
+load("@io_tweag_rules_haskell//haskell:c2hs.bzl", "c2hs_toolchain")
```

Comment:
The majority of the changes were related to moving the c2hs_toolchain into its own import